### ISSUE TITLE
Better settings than ones of original paper.

### ISSUE THREAD

**Issue by Hzzone** [dawei6875797/Face-Aging-with-Identity-Preserved-Conditional-Generative-Adversarial-Networks#21]
In my experiments, I found the following:
1. With the 512 final output channels of PatchDiscriminator, the generator will easily suffer from mode collapse or gradient explosion. In the original paper, they set the epsilon of batch norm to 0.001 and thus solved this problem.
2. However, the initial PatchGAN[Image-to-Image Translation with Conditional Adversarial Networks] has the 1 final output channel to make sure every $N\times N$ patch to be real or fake. The original paper set 1 to 512 will make the training process unstable because of the above issues.

Therefore, I recommended that the final output set to 1 and the epsilon of batch norm to 1e-5 which is the default setting of PyTorch. The experiment shows that my setting has successfully reproduced the result of the original paper using the same hyperparameters.

Hopefully, it helps the future works of Face Aging.

**Comment by Hzzone** [comment]
Moreover, if your input image size is larger than the original one, 128, please increase the downsampling convolution layers. For example, 4 stride-2 convolution layers for 256 input size.

**Comment by ThanhBarTN** [comment]
> Moreover, if your input image size is larger than the original one, 128, please increase the downsampling convolution layers. For example, 4 stride-2 convolution layers for 256 input size.

Thanks man that helps alot
